broadcaster/_backends/redis.py

```diff
@@ -1,34 +1,41 @@
-import asyncio_redis
+import aioredis
+import asyncio
 import typing
-from urllib.parse import urlparse
+
 from .base import BroadcastBackend
 from .._base import Event
 
 
 class RedisBackend(BroadcastBackend):
     def __init__(self, url: str):
-        parsed_url = urlparse(url)
-        self._host = parsed_url.hostname or "localhost"
-        self._port = parsed_url.port or 6379
+        self.conn_url = url
+        self.channel = None
 
     async def connect(self) -> None:
-        self._pub_conn = await asyncio_redis.Connection.create(self._host, self._port)
-        self._sub_conn = await asyncio_redis.Connection.create(self._host, self._port)
-        self._subscriber = await self._sub_conn.start_subscribe()
+        loop = asyncio.get_event_loop()
+        self._pub_conn = await aioredis.create_redis(self.conn_url, loop=loop)
+        self._sub_conn = await aioredis.create_redis(self.conn_url, loop=loop)
 
     async def disconnect(self) -> None:
         self._pub_conn.close()
         self._sub_conn.close()
 
     async def subscribe(self, channel: str) -> None:
-        await self._subscriber.subscribe([channel])
+        channel = await self._sub_conn.subscribe(channel)
+        self.channel = channel[0]
 
     async def unsubscribe(self, channel: str) -> None:
-        await self._subscriber.unsubscribe([channel])
+        await self._sub_conn.unsubscribe(channel)
 
     async def publish(self, channel: str, message: typing.Any) -> None:
-        await self._pub_conn.publish(channel, message)
+        await self._pub_conn.publish_json(channel, message)
+
+    async def _wait_for_channel(self) -> None:
```

Comment:
I would try to find a way to avoid this. With async-redis it was possible, so I guess it should be possible to do it with aioredis, what do you think?

Reply:
Thanks, @Woile. It would be good to find a solution because we have the same problem in `redis-stream` backend.

Maybe we need to change the way of subscription on the base class. what do you think?

Reply:
I have moved listener_task creation from connect to subscribe. now we don't need the `_wait_for_channel` function.